Update install instructions

pllim opened the pull request:
Fix #72 . Anaconda full install is overkill, so I also changed the instructions to point to Miniconda.

Here is an example `check_env.py` output from an env created with this patch using `conda` 4.7.12:

```
(astropy-workshop) % python check_env.py 
Found IPython 7.9.0
Found cython 0.29.13
Found jupyter 
Found notebook 6.0.2
Found numpy 1.17.3
Found scipy 1.3.1
Found skimage 0.15.0
Found matplotlib 3.1.1
Found pandas 0.25.3
Found bs4 4.8.1
Found keyring 
Found html5lib 1.0.1
Found xlwt 1.3.0
Found requests 2.22.0
Found jupyterlab 2.0.0a1
Found astropy 4.0rc1
Found asdf 2.4.2
Found gwcs 0.11.0
Found photutils 0.7.1
Found specutils 0.6
Found glue_vispy_viewers 0.12.2
Found glue 0.15.6
Found astroquery 0.4.dev0

Your Python environment is good to go!
```

pllim commented:
Do we also need to add `git` to this list? Or do you rely on people installing Git on their own?

stargaser commented:
> Do we also need to add git to this list? Or do you rely on people installing Git on their own?

Oh, I had not realized `git` could be installed with conda. I think I would prefer not to include it in `environment.yml`  but we could point out it can be installed into the environment with conda.

I've paged through the file changes. I want to try out the new instructions before completing my review.

stargaser commented on 00-Install_and_Setup/environment.yml:
```diff
@@ -4,28 +4,29 @@ channels:
+  - scikit-image>=0.14
+  - matplotlib>=2.2.3
+  - pandas>=0.23
   - pip:
```
When I installed the environment, I got a warning about pip not being in the list of conda-installed packages. It worked anyway, but maybe we want to avoid the warning by adding pip.

pllim replied:
I added it. Is that what you meant?

stargaser replied:
Yes, thanks!

eteq commented on 00-Install_and_Setup/README.md:
```diff
@@ -1,77 +1,125 @@
 
-*Anaconda includes conda, conda build, Python and 100+ automatically installed, open source scientific packages and their dependencies that have been tested to work well together, including SciPy, NumPy and many others.*
+*Miniconda is a free minimal installer for conda. It is a small, bootstrap
+version of Anaconda that includes only conda, Python, the packages they depend
```
(note somewhere in here that anaconda can be used if already installed)

pllim replied:
Is this already handled by your suggestion at a8ecb5a3108fb194f01a3c438f974ec1e9943b17 ?

stargaser replied:
I think it's covered.

eteq commented on 00-Install_and_Setup/README.md:
```diff
@@ -1,77 +1,125 @@
 Please come to the workshop with a laptop already configured as described below.
-If you have any problems with any of these steps, please [open an issue](https://github.com/astropy/astropy-workshop/issues).
+If you have any problems with any of these steps, please
+[open an issue](https://github.com/astropy/astropy-workshop/issues).
```
maybe
```suggestion
[ask your question in the issue tracker](https://github.com/astropy/astropy-workshop/issues/new?assignees=&labels=workshop-question&template=question-from-workshop-participant.md&title=%5BQuestion%5D+Summarize+your+question+here).
```
? Unless we want them to see the issues list first...

eteq commented on 00-Install_and_Setup/README.md:
```diff
@@ -1,77 +1,125 @@
+*Miniconda is a free minimal installer for conda. It is a small, bootstrap
+version of Anaconda that includes only conda, Python, the packages they depend
+on, and a small number of other useful packages, including pip, zlib and a few
+others.*
```
```suggestion
others. Note, though, that if you have either miniconda or the full Anaconda 
already installed, you can skip to the next step.*
```

eteq commented on 00-Install_and_Setup/README.md:
```diff
@@ -1,77 +1,125 @@
 
-[Create a conda environment for this workshop using a yml file](https://conda.io/docs/user-guide/tasks/manage-environments.html#creating-an-environment-from-an-environment-yml-file). 
-The python version and all needed packages, including astropy, are listed in the [environment.yml](https://github.com/astropy/astropy-workshop/blob/master/00-Install_and_Setup/environment.yml) file
+*Miniconda includes an environment manager called conda. You can create,
```
Maybe should include a definition of what an "environment" is?  Something like:
```suggestion
*Miniconda includes an environment manager called conda. Environments
allow you to have multiple sets of Python packages installed at the same 
time, making reproducibility and upgrades easier. You can create,
```

pllim commented:
I think I have addressed all review comments? As for "should we really do all of this," I would defer to you all. Please let me know if further changes are necessary. Otherwise, please merge so we can proceed with the next step of planning. Thanks!
